build-logic/src/main/kotlin/CompilerOptions.kt

```diff
@@ -59,6 +60,15 @@ fun Project.configureJavaAndKotlinCompilers() {
   allWarningsAsErrors(true)
 }
 
+fun Project.configureTests(jvmVersion: Int) {
+  tasks.withType(Test::class.java).configureEach {
+    val javaToolchains = this@configureTests.extensions.getByName("javaToolchains") as JavaToolchainService
```

Comment:
also here you could probably use `configure<JavaToolchainService> { }` syntax (https://gradle.github.io/kotlin-dsl-docs/api/org.gradle.kotlin.dsl/org.gradle.api.plugins.-extension-aware/configure.html)

e: ah I see it's `build-logic`. FYI you can also use Kotlin DSL extensions by adding a regular dependency  `implementation(org.gradle.kotlin.dsl.ProjectExtensionsKt.gradleKotlinDsl(project))`, and `"org.jetbrains.kotlin.plugin.sam.with.receiver"` plugin (needed to avoid the `it`s in lambdas)

Reply:
I'm on team `no kotlin-dsl`. I know I'm the outlier there but there are enough different ways to do the same things in Gradle that I don't need a `fun String.invoke(dependencyNotation: Any)` in the classpath. 

That's a long story but I believe a good proportion of the pain Kotlin devs have using Kotlin build scripts is because we're trying to hard to make it look like a DSL when in fact it's not and the logic behind is imperative. The order of calls matters, etc..

Reply:
Oh, I'm similar camp but I guess with slightly different approach — I'm 100% against `gradle.kts`  and IMO if only `gradleKotlinDsl()` API existed it'd just be a nice sugar striking a good balance between `val check by tasks.existing` and using pure Java APIs and manual casting, and it would be a nice addition to build logic writte in Kotlin. Agreed it sometimes goes too far though, `String.invoke` _is_ :unamused: 

Anyway, just wanted to share that `gradleKotlinDsl` dependency trick because it took me a while to find it :slightly_smiling_face: